lib/sendgrid/recipient.rb

```diff
@@ -0,0 +1,26 @@
+require 'smtpapi'
+
+module SendGrid
+  class Recipient
+
+    attr_reader :address, :substitutions
+
+    def initialize(address)
+      @address = address
+      @substitutions = {}
+    end
+
+    def add_substitution(key, value)
+      substitutions[key.to_sym] = value
+    end
+
+    def add_to_smtpapi(smtpapi)
+      return if @address.nil? || @substitutions.empty?
```

Comment:
We should ensure address is not nil elsewhere, as well as ensure substitutions has content.  It seems that this object won't function correctly in a whole bunch of states that are extremely easy to access for a user.

Also, it seems like the responsibility of this object is to hide the lower-level smptapi object, but it's still fully exposed to the client class.